File: IndieLibX/Samples/2DAnimation/2DAnimation.cpp
/*
  Copyright (c) 2015 Tommi Roenty   http://www.tommironty.fi/
  Licensed under The GNU Lesser General Public License, version 2.1:
      http://opensource.org/licenses/LGPL-2.1
*/
#include <Framework3/IRenderer.h>

const float _size=50.0f;
const char* name[] =
{
"mario_stand", 
"mario_walk", 
"Mario_jump",
0
};

class _2DAnimation : public STXGUI
{
public:
	_2DAnimation()
	{
		anim=0;
	}
	virtual ~_2DAnimation()
	{
		delete anim;
	}
public:
	FW3_Animation* anim;

virtual void actionPerformed(GUIEvent &evt)
{
  const NSString &callbackString  = evt.getCallbackString();
  GUIRectangle *sourceRectangle = evt.getEventSource();
  int           widgetType      = sourceRectangle->getWidgetType();

  if(widgetType == RADIO_BUTTON)
  {
    GUIRadioButton   *radioButton = (GUIRadioButton*)sourceRectangle;
    if(radioButton->isPressed() && radioButton->isChecked())
    {
	unsigned int i=0;
	while(1)
	{
		if(name[i]==0)
			break;
      		if(callbackString == name[i])
      		{
			anim->SetSequence(name[i]);
			break;
		}
		i++;
	}
    }
  }
}
int init(const char* aTitle)
{
	anim=new FW3_Animation("/2DAnimation/mario.xml");
	anim->SetSequence("mario_stand");
	//anim->SetSequence("mario_walk");
	//anim->SetSequence("Mario_jump");

	anim->SetPosition(
		(IRenderer::GetRendererInstance()->GetWidth()-_size)/2,
		(IRenderer::GetRendererInstance()->GetHeight()-_size)/2);
	anim->SetSize(_size,_size);
	STXGUI::init("/2DAnimation/GUILayout.xml");
	return 0;
}
void render()
{
	IRenderer::GetRendererInstance()->Clear( true,true, D3DXFROMWINEVECTOR4(0.35f, 0.53f, 0.7, 1.0f));
	IRenderer::GetRendererInstance()->BeginScene();

	anim->render();
	STXGUI::update();

	IRenderer::GetRendererInstance()->EndScene();
	IRenderer::GetRendererInstance()->Present( );
}
};
int ApplicationLogic()
{
	IRenderer* r=IRenderer::GetRendererInstance("2DAnimation");		
	IInput*    i=STX_Service::GetInputInstance();
	_2DAnimation app;
	app.init("");
	while (!i->OnKeyPress (STX_KEY_ESCAPE) && !i->Quit())
	{
		i->Update();	
		app.render();
	}
	return 0;
}

File: IndieLibX/scripting/2DAnimation.py
#
# Copyright (c) 2015 Tommi Roenty   http://www.tommironty.fi/
# Licensed under The GNU Lesser General Public License, version 2.1:
#     http://opensource.org/licenses/LGPL-2.1
#
import IndieLib

class AppPy:
	def __init__(self):
		self.name=["mario_stand", "mario_walk", "Mario_jump"]

	def actionPerformed(self):
		IndieLib.UpdateNSGUISWIG()
		loop=not IndieLib.isGUIEventsEmpty()
		while loop:
			evt=IndieLib.PollGUIEvent()
			loop=not IndieLib.isGUIEventsEmpty()
			if evt:
				callbackString=evt.getCallbackCString()
				if callbackString=="Unknown":
					continue
				if IndieLib.isClicked(evt):
					for i in range(0,len(self.name)):
      						if callbackString == self.name[i]:
							#print i
							self.anim.SetSequence(self.name[i])
							break

	def init(self, width, height):
		self.anim=IndieLib.FW3.GetInstance().GetAnimation("/2DAnimation/mario.xml")

		IndieLib.InitNSGUISWIG("/2DAnimation/GUILayout.xml")

		self.anim.SetSequence("mario_stand")
		#self.anim.SetSequence("mario_walk")
		#self.anim.SetSequence("Mario_jump")
		self.anim.SetPosition(IndieLib.IRenderer.GetRendererInstance().GetWidth()/2-25.0, IndieLib.IRenderer.GetRendererInstance().GetHeight()/2-25.0)
		self.anim.SetSize(50.0,50.0)
	def render(self):
		IndieLib.IRenderer.GetRendererInstance().Clear( True, True, IndieLib.D3DXFROMWINEVECTOR4(0.35, 0.53, 0.7, 1.0))
		IndieLib.IRenderer.GetRendererInstance().BeginScene()
		self.actionPerformed()

		self.anim.render()

		IndieLib.IRenderer.GetRendererInstance().EndScene()
		IndieLib.IRenderer.GetRendererInstance().Present( )

if __name__=="__main__":
	app=AppPy()
	IndieLib.IRenderer.GetRendererInstance("2DAnimation")
	app.init(640, 512)
	while ((not IndieLib.STX_Service.GetInputInstance().OnKeyPress (IndieLib.STX_KEY_ESCAPE)) and (not IndieLib.STX_Service.GetInputInstance().Quit())):
		IndieLib.STX_Service.GetInputInstance().Update()	
		app.render()

